### PR TITLE
CAP-171 Add data access code for AURN site data

### PR DESCRIPTION
**Pull request by crbunney**
Added data access code for AURN site data file stored on object store, plus supporting classes/functions.

Documented in clean_air/data/static_aurn_site_data.md

### CODE REVIEW

**Comment by corinnebosley** [comment]
Well there seem to be 15 test fails for this PR, all with the same error message: 

An error occurred (InvalidAccessKeyId) when calling the CreateBucket operation: The AWS Access Key Id you provided does not exist in our records.

File: README.md
Comment (corinnebosley) [inline README.md:25]: Is this TODO the reason for all the test fails (credentials aren't there yet, so of course they don't exist in the thing)?

If so, then I would be happy to merge this PR regardless of fails.

**Review by corinnebosley — COMMENTED** [review]
This all looks pretty good to me, except for the access key errors.  I especially love the full explanations of everything within your code, you write beautifully, if I may say so.

I would like your tests to pass before I merge this, but if you don't consider that possible then let me know and I will approve and merge.

File: README.md
Comment (crbunney) [reply README.md:25]: That was more that I wasn't sure how credentials needed to be stored. I could probably put a bit about `boto3` in here, but I'm not sure how the `s3fs` based stuff gets its credentials. It might be the same as it's using `botocore`, but I hadn't looked into it to be sure

**Comment by crbunney** [comment]
> you write beautifully, if I may say so.

Thank you :)

> I would like your tests to pass before I merge this, but if you don't consider that possible then let me know and I will approve and merge.

That's basically due to this issue: https://github.com/spulec/moto/issues/2993
It's not picking up the correct version of the library, probably because it's not been properly release yet. In my env I installed it from their branch on git, so I guess I need to find another way of specifying it in the `environment.yml`. I'll do that

**Comment by corinnebosley** [comment]
Weirdly pip seems to have a problem with line 23 of the setup.cfg, it seems to be claiming that it already exists further up in the file.  That seems odd, because this is line 23:
`[options.packages.find]`

**Comment by crbunney** [comment]
> Weirdly pip seems to have a problem with line 23 of the setup.cfg, it seems to be claiming that it already exists further up in the file. That seems odd, because this is line 23: `[options.packages.find]`

Yea, this is a fun one. Can't reproduce it locally, both `conda env create -f environment.yml --name tmp` and `pip install /path/to/folder` work fine

**Comment by crbunney** [comment]
This is very peculiar, as the build is failing during creation of the conda environment. The conda environment (as far as I know) shouldn't try to install the project, as it's not referenced in the `environment.yml`, which implies it's not failing because of that.

However, even with verbosity cranked up, I can't see exactly what conda is trying to install via pip because the verbosity setting doesn't affect the pip output (see https://github.com/conda/conda/issues/10556). Also, it passes a temporary requirements.txt file to pip, which is cleaned up after the command exits, so I can't see what's being passed into the command either.

Mainly I've been trying to explore the possibility that it's a problem with a dependency that's perhaps introduced a bug into the most recent release.

**Comment by corinnebosley** [comment]
Maybe we should take this opportunity to tidy up our `environment.yml`.  There are definitely several dependencies which can be removed (datashader, holoviews, geoviews) and I was going to do this some time soon anyway.

It probably won't solve the problem but it might at least make it quicker for the environment build to fail.

**Comment by crbunney** [comment]
Fun fun fun.

Understanding why this was failing requires knowledge of three facts:
1) The diff shown in GitHub's pull request UI is the difference between the start of the branch and the latest commit (i.e. the **unmerged** branch)
2) The github action trigger we're using (`pull_request`) defaults to running on the **merged** code (i.e. what the code would look like if the branch was merged into the target branch)
3) Barnaby spotted the same problem as I did and fixed it in some work that got merged last week

So basically what happened is that, despite having no conflicts, merging my branch into the current master resulted in an invalid setup.cfg that had the `package` element repeated. It was sheer coincidence that the repeated element happened to be on line 23 in the merged file, which gave the false impression of a problem in a different part of the unmerged file.

Removing my edits to the packaging setup got rid of the duplicated element and allowed the build to pass

**Review by corinnebosley — APPROVED** [review]
I can see that you've worked very hard on this PR, and I am very impressed with your storage tests, very nice.

I am happy with this PR, and I am very well aware of how painful it must have been fixing the weird test errors.  I doubt Catherine will have the time to review and merge this, so I am happy to approve and merge it myself.

**Pull request merged**
